src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,85 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import cn from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { ResponseError } from '../../types/ResponseError';
+import { MovieData } from '../../types/MovieData';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+
+const DEFAULT_POSTER
+  = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+type Props = {
+  addMovie: (film: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [data, setData] = useState<MovieData | null | ResponseError>(null);
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+
+  const handleQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+    if (isError) {
+      setIsError(false);
+    }
+  };
+
+  const findMovie = useCallback(() => {
+    setIsLoading(true);
+    getMovie(query.trim())
+      .then(movieData => {
+        setData(movieData);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [query]);
+
+  const handleFindButton = () => {
+    findMovie();
+  };
+
+  const handleInputKeyPressEnter = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (e.key !== 'Enter') {
+      return;
+    }
+
+    e.preventDefault();
+    findMovie();
+  };
+
+  const handleAddButton = () => {
+    addMovie(foundMovie as Movie);
+    setQuery('');
+    setFoundMovie(null);
+  };
+
+  useEffect(() => {
+    if (data !== null && 'Title' in data) {
+      const newMovie: Movie = {
+        title: data.Title,
+        description: data.Plot,
+        imgUrl: (!data.Poster || data.Poster === 'N/A')
+          ? DEFAULT_POSTER
+          : data.Poster,
+        imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
```

Comment:
it's better to move 'https://www.imdb.com/title/' to the constant, because it's a static data